Tools/TestWebKitAPI/Scripts/process-entitlements.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+set -e
+
+function plistbuddy()
+{
+    /usr/libexec/PlistBuddy -c "$*" "${WK_PROCESSED_XCENT_FILE}"
+}
+
+function plistbuddy_base()
+{
+    if [[ "${WK_PLATFORM_NAME}" =~ .*simulator ]]
+    then
+        /usr/libexec/PlistBuddy -c "$*" "${WK_PROCESSED_BASE_XCENT_FILE}"
+    else
+        plistbuddy "$*"
+    fi
+}
+
+# ========================================
+# Common entitlements
+# ========================================
+
+process_base_testwebkitapi_entitlements()
+{
+    if [[ "${CONFIGURATION}" != Production ]]
+    then
+        plistbuddy_base Add :com.apple.security.get-task-allow bool YES
+    fi
+}
+
+process_restricted_testwebkitapi_entitlements()
+{
+    plistbuddy Add :keychain-access-groups array
+    plistbuddy Add :keychain-access-groups:0 string com.apple.TestWebKitAPIAlternate
+    plistbuddy Add :keychain-access-groups:1 string com.apple.TestWebKitAPI
+    plistbuddy Add :com.apple.private.webkit.adattributiond.testing bool YES
+    plistbuddy Add :com.apple.private.webkit.webpush bool YES
+    plistbuddy Add :com.apple.private.webkit.webpush.inject bool YES
+}
+
+# ========================================
+# Mac entitlements
+# ========================================
+
+function process_mac_testwebkitapi_entitlements()
+{
+    process_base_testwebkitapi_entitlements
+
+    plistbuddy Add :com.apple.security.temporary-exception.sbpl array
+    plistbuddy Add :com.apple.security.temporary-exception.sbpl:0 string '(allow mach-issue-extension (require-all (extension-class \"com.apple.webkit.extension.mach\")))'
+    plistbuddy Add :com.apple.security.temporary-exception.sbpl:1 string '(allow iokit-issue-extension (require-all (extension-class \"com.apple.webkit.extension.iokit\")))'
+
+    if [[ "${WK_USE_RESTRICTED_ENTITLEMENTS}" == YES ]]
+    then
+        process_restricted_testwebkitapi_entitlements
+
+        plistbuddy Add :com.apple.hid.manager.user-access-device bool YES
+        plistbuddy Add :com.apple.private.hid.client.event-filter bool YES
+    fi
+}
+
+# ========================================
+# iOS-Family Device entitlements
+# ========================================
+
+function process_ios_family_testwebkitapi_entitlements()
+{
+    process_base_testwebkitapi_entitlements
+    process_restricted_testwebkitapi_entitlements
+
+    plistbuddy Add :com.apple.developer.WebKit.ServiceWorkers bool YES
+    plistbuddy Add :com.apple.Pasteboard.paste-unchecked bool YES
+    plistbuddy Add :com.apple.private.xpc.launchd.job-manager string TestWebKitAPI
+    plistbuddy Add :com.apple.CommCenter.fine-grained array
+    plistbuddy Add :com.apple.CommCenter.fine-grained:0 string public-cellular-plan
+}
+
+rm -f "${WK_PROCESSED_XCENT_FILE}"
+rm -f "${WK_PROCESSED_BASE_XCENT_FILE}"
+plistbuddy Clear dict
+plistbuddy_base Clear dict
```

Comment:
nit: Do you really need to rm -f here? Seems like `PlistBuddy Clear ...` will reset the plist for you.